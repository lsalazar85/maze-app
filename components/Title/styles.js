import styled from "styled-components";

export const TitleElement = styled.h1`
    margin-bottom: 3rem;
`