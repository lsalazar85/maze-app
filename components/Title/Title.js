import { string } from "prop-types";
import { TitleContainer, H1Element } from "./styles";

const Title = ({ title }) => (
    <TitleContainer>
        <H1Element>{title}</H1Element>
    </TitleContainer>
)

Title.propTypes = {
    title: string.isRequired,
}

export default Title