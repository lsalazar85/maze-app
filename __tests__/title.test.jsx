/**
 * @jest-environment jsdom
 */

import {render, screen} from '@testing-library/react'
import Title from "@/components/Title";


describe('Maze', () => {
    const text = 'Maze Game'

    it('should render correctly', () => {
        const { asFragment } = render(<Title title={text} />)

        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    })

    it('component render without props', () => {
        const { asFragment } = render(<Title />)

        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    })

    it('should render correctly with title', () => {
        render(<Title title={text} />)

        const title = screen.getByText(/maze game/i)
        expect(title).toBeInTheDocument()
    })
})