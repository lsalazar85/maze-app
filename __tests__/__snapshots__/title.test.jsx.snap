// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maze component render without props 1`] = `
<DocumentFragment>
  <div
    class="styles__TitleContainer-sc-16jy8ol-0 epXWgE"
  >
    <h1
      class="styles__H1Element-sc-16jy8ol-1 exeysa"
    />
  </div>
</DocumentFragment>
`;

exports[`Maze should render correctly 1`] = `
<DocumentFragment>
  <div
    class="styles__TitleContainer-sc-16jy8ol-0 epXWgE"
  >
    <h1
      class="styles__H1Element-sc-16jy8ol-1 exeysa"
    >
      Maze Game
    </h1>
  </div>
</DocumentFragment>
`;
